cli/lsp/urls.rs

```diff
@@ -103,12 +103,32 @@ impl LspUrlMap {
   }
 
   /// Normalize URLs from the client, where "virtual" `deno:///` URLs are
-  /// converted into proper module specifiers.
+  /// converted into proper module specifiers, and handle situations where file
+  /// URLs on windows can be canonicalized differently.
   pub fn normalize_url(&self, url: &Url) -> ModuleSpecifier {
     if let Some(specifier) = self.get_specifier(url) {
       specifier.clone()
     } else {
-      url.clone()
+      let mut url = url.clone();
+      if url.scheme() == "file" && url.as_str().contains("%3A") {
+        let path = url
+          .path_segments()
+          .unwrap()
+          .map(|s| {
+            if s.ends_with("%3A") && s.len() == 4 {
```

Comment:
* Suggestion: apply only to the first path segment, and only if the url has no (/empty) hostname.
* I don't quite understand why this is necessary. Does "generic" normalization not apply to `%3A`.
* What about `file:///c%3a/foo` (lowecase a) and `file:///c|/foo` (alternative notation) ?